File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(number) {
    this.number =number;
    this.SumOfSquares=0;
    this.SquareOfSum=0; 
  }

  get sumOfSquares() {
    for(let i=0; i<=this.number; i++){
      this.SumOfSquares+= Math.pow(i, 2)
    }
    return this.SumOfSquares
  }  

  get squareOfSum() {
    for(let i=0; i<=this.number; i++){
      this.SquareOfSum+=i
    }
    this.SquareOfSum= Math.pow(this.SquareOfSum, 2) 
    return this.SquareOfSum
  }
    
  get difference() {
    return  this.SquareOfSum - this.SumOfSquares 
  }
}

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
    def initialize(number)
        @number = number
    end

    def square_of_sum
        (1..@number).to_a.sum**2 
       
    end

    def sum_of_squares
        (1..@number).to_a.map{|el| el **2}.sum
    end

    def difference
        square_of_sum - sum_of_squares
    end
end

